numpy/core/arrayprint.py

```diff
@@ -395,9 +399,13 @@ def _object_format(o):
     return fmt.format(o)
 
 def repr_format(x):
+    if isinstance(x, (np.str_, np.bytes_)):
```

Comment:
On the one hand, yeah. On the other hand, this is the drop-through for user defined dtypes and I don't know if it is wise to call `.item()` on those.

Reply:
Ah, good catch, I had not realized that `numpystr` is used as a default for `dtype.type is None`.

Since really the point of my earlier comments was to keep the changes minimal, it makes sense to stick with your choice - though perhaps add a TODO that really this is to be fixed? 

Also, would it work to make the change in `repr_format`? Is `str_format` used? Indeed, for `str_format`, is there actually any difference between `str(x)` and `str(x.item())`?